lib/output.js

```diff
@@ -72,47 +86,36 @@ function OutputStream(options) {
     }, true);
 
     // Convert comment option to RegExp if neccessary and set up comments filter
-    if (typeof options.comments === "string" && /^\/.*\/[a-zA-Z]*$/.test(options.comments)) {
-        var regex_pos = options.comments.lastIndexOf("/");
-        options.comments = new RegExp(
-            options.comments.substr(1, regex_pos - 1),
-            options.comments.substr(regex_pos + 1)
-        );
-    }
-    if (options.comments instanceof RegExp) {
-        options.comments = (function(f) {
-            return function(comment) {
-                return comment.type == "comment5" || f.test(comment.value);
-            }
-        })(options.comments);
-    }
-    else if (typeof options.comments === "function") {
-        options.comments = (function(f) {
-            return function(comment) {
-                return comment.type == "comment5" || f(this, comment);
-            }
-        })(options.comments);
-    }
-    else if (options.comments === "some") {
-        options.comments = function(comment) {
-            var text = comment.value;
-            var type = comment.type;
-            if (type == "comment2") {
-                // multiline comment
-                return /@preserve|@license|@cc_on/i.test(text);
-            }
-            return type == "comment5";
+    var comment_filter;
```

Comment:
There is no point in improving code logic when architecturally the code didn't make sense. Also, the code fix was up in about 1 hour after the report. Which may have (and did) contributed to lower quality of the patch.